src/main/java/com/minecolonies/coremod/event/EventHandler.java

```diff
@@ -309,6 +314,22 @@ else if ("pmardle".equalsIgnoreCase(event.getEntityPlayer().getName())
             {
                 return;
             }
+             //Checks to see if player tries to sleep in a bed belonging to a Citizen, ancels the event, and Notifies Player that bed is occuppied
+            if(world.getBlockState(event.getPos()).getBlock().isBed(world.getBlockState(event.getPos()), world, event.getPos(), player))
+            {
+                BlockPos bd = event.getPos();
+                if(world.getBlockState(event.getPos()).getBlock().isBedFoot(world, event.getPos())) {
+                    bd = bd.offset(world.getBlockState(event.getPos()).getValue(BlockBed.FACING));
+                }
+                //Searches through the nearest Colonie's Citizen and sees if the bed belongs to a Citizen, and if the Citizen is asleep
+                final ArrayList<CitizenData> cits = new ArrayList<>(ColonyManager.getClosestColony(world, event.getPos()).getCitizenManager().getCitizens());
```

Comment:
List<CitizenData> cits = new ArrayList<>();
Also, better save the colony and check for null first, if colony is null no need for all of this.